examples/Atlas/+atlasUtil/plotWalkingTraj.m

```diff
@@ -23,8 +23,24 @@
 dtraj = fnder(PPTrajectory(spline(tts,x_smooth)));
 qddtraj = dtraj(nq+(1:nq));
 
-lfoot = findLinkId(r,'l_foot');
-rfoot = findLinkId(r,'r_foot');
+for j = 1:length(walking_plan_data.body_motions)
+  if walking_plan_data.body_motions(j).body_id == r.foot_body_id.right || ...
+     (walking_plan_data.body_motions(j).body_id < 0 && r.getFrame(walking_plan_data.body_motions(j).body_id).body_ind == r.foot_body_id.right)
+    r_ind = j;
+  elseif  walking_plan_data.body_motions(j).body_id == r.foot_body_id.left || ...
+     (walking_plan_data.body_motions(j).body_id < 0 && r.getFrame(walking_plan_data.body_motions(j).body_id).body_ind == r.foot_body_id.left)
+    l_ind = j;
+  end
+end
+
+rfoot_motion = walking_plan_data.body_motions(r_ind);
+if rfoot_motion.body_id < 0
+  rfoot_body = r.getFrame(rfoot_motion.body_id).body_ind;
+end
+lfoot_motion = walking_plan_data.body_motions(l_ind);
+if lfoot_motion.body_id < 0
+  lfoot_body = r.getFrame(lfoot_motion.body_id).body_ind;
+end
```

Comment:
Lots of code duplication between left and right here. Would be better to use a for loop.